src/Filters/design.jl

```diff
@@ -569,19 +594,31 @@ FIRWindow(; transitionwidth::Real=throw(ArgumentError("must specify transitionwi
     FIRWindow(kaiser(kaiserord(transitionwidth, attenuation)...), scale)
 
 # Compute coefficients for FIR prototype with specified order
-function firprototype(n::Integer, ftype::Lowpass, fs::Real)
+function _firprototype(n::Integer, ftype::Lowpass, fs::Real, ::Type{T}) where {T<:Number}
     w = normalize_freq(ftype.w, fs)
 
-    [w*sinc(w*(k-(n+1)/2)) for k = 1:n]
+    promote_type(typeof(w), T)[w*sinc(w*(k-(n+1)/2)) for k = 1:n]
 end
 
+firprototype(n::Integer, ftype::Lowpass, fs::Real) =
+    _firprototype(n, ftype, fs, typeof(fs))
+
 function firprototype(n::Integer, ftype::Bandpass, fs::Real)
     w1 = normalize_freq(ftype.w1, fs)
     w2 = normalize_freq(ftype.w2, fs)
 
     [w2*sinc(w2*(k-(n+1)/2)) - w1*sinc(w1*(k-(n+1)/2)) for k = 1:n]
 end
 
+function firprototype(n::Integer, ftype::ComplexBandpass, fs::T) where {T<:Real}
+    w1 = normalize_complex_freq(ftype.w1, fs)
+    w2 = normalize_complex_freq(ftype.w2, fs)
+    w_center = (w2 + w1) / 2
+    w_cutoff = (w2 - w1) / 2
+    lp = Lowpass(w_cutoff)
+    _firprototype(n, lp, fs, Complex{T}) .*= cispi.(w_center * (0:(n-1)))
```

Comment:
As `w_cutoff` is a normalized frequency, we have to use `fs=2` for it:
```suggestion
    _firprototype(n, lp, 2, Complex{T}) .*= cispi.(w_center * (0:(n-1)))
```
But I suspect this will make the test fail.